src/services/authorizedItem.service.ts

```diff
@@ -136,26 +136,141 @@ export class AuthorizationService {
     return { itemMemberships: resultOfMemberships, visibilities };
   }
 
+  /**
+   * Returns whether the actor has a membership on the item and can access it.
+   * Having an admin membership returns true.
+   * Having a write membership returns true.
+   * Having a read membership and the item is hidden returns false.
+   * Having a read membership and the item is public returns true.
+   * Having no membership and public returns false.
+   * @returns whether the member has permission and can access the item
+   */
   public async hasPermission(
     dbConnection: DBConnection,
-    permission: PermissionLevelOptions,
-    actor: MaybeUser,
-    item: ItemRaw,
+    {
+      permission = PermissionLevel.Read,
+      actor,
+      item,
+    }: { permission?: PermissionLevelOptions; actor: MaybeUser; item: ItemRaw },
   ) {
     try {
-      await this.validatePermission(dbConnection, permission, actor, item);
+      const { itemMembership, visibilities } = await this.getItemWithProperties(dbConnection, {
+        permission,
+        actor,
+        item,
+      });
+
+      // returns false if does not have membership and item is public
+      if (!itemMembership && visibilities.find(({ type }) => type === ItemVisibilityType.Public)) {
+        return false;
+      }
+
       return true;
-    } catch (err: unknown) {
+    } catch (_e) {
       return false;
     }
   }
 
-  public async validatePermission(
+  /**
+   * Returns whether the actor can access an item.
```

Comment:
It does not look like it returns anything. Update phrasing to make it clear that the function passes if the actor has rights, if not, it throws.